MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.cpp

```diff
@@ -0,0 +1,102 @@
+                  including <type>BrookCorey</type>
+                  and it has a tag of <capillary_pressure>
+*/
+static std::unique_ptr<CapillaryPressureSaturation>
+                         createBrookCorey(BaseLib::ConfigTree const& config)
+{
+    std::array<double, 5> parameters = {
+        {//! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pd}
+         config.getConfigParameter<double>("pd"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__sr}
+         config.getConfigParameter<double>("sr"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__smax}
+         config.getConfigParameter<double>("smax"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__m}
+         config.getConfigParameter<double>("m"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pc_max}
+         config.getConfigParameter<double>("pc_max")}};
+
+    assert(parameters[3] >= 1.0);  // m >= 1
+    return std::unique_ptr<CapillaryPressureSaturation>(
+                        new BrookCoreyCapillaryPressureSaturation(parameters));
+}
+
+/**
+    \param config ConfigTree object which contains the input data
+                  including <type>vanGenuchten</type>
+                  and it has a tag of <capillary_pressure>
+*/
+static std::unique_ptr<CapillaryPressureSaturation>
+                         createVanGenuchten(BaseLib::ConfigTree const& config)
+{
+    std::array<double, 5> parameters = {
+        {//! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pd}
+         config.getConfigParameter<double>("pd"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__sr}
+         config.getConfigParameter<double>("sr"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__smax}
+         config.getConfigParameter<double>("smax"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__m}
+         config.getConfigParameter<double>("m"),
+         //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type__pc_max}
+         config.getConfigParameter<double>("pc_max")}};
+
+    assert(parameters[3] <= 1.0);  // m <= 1
+    return std::unique_ptr<CapillaryPressureSaturation>(
+                      new vanGenuchtenCapillaryPressureSaturation(parameters));
+}
+
+std::unique_ptr<CapillaryPressureSaturation> createCapillaryPressureModel(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__capillary_pressure__type}
+    auto const type = config.getConfigParameter<std::string>("type");
```

Comment:
Note: If you'd add checks to the specific create functions, you'd switch to `peek...()` here.